compiler/rustc_mir_transform/src/check_alignment.rs

```diff
@@ -30,6 +29,7 @@ impl<'tcx> MirPass<'tcx> for CheckAlignment {
 
         let basic_blocks = body.basic_blocks.as_mut();
         let local_decls = &mut body.local_decls;
+        let param_env = tcx.param_env_reveal_all_normalized(body.source.def_id());
 
         for block in (0..basic_blocks.len()).rev() {
```

Comment:
The interaction between the iteration order and what `split_block` does definitely needs a comment.